### PR TITLE
Adding service parameter to GetentUtils

### PR DESCRIPTION
**Pull request by sidecontrol**

### CODE REVIEW

File: sssd_test_framework/utils/tools.py [hunk 1]
```diff
@@ -541,41 +541,49 @@ def __init__(self, host: MultihostHost) -> None:
         """
         super().__init__(host)
 
-    def passwd(self, name: str | int) -> PasswdEntry | None:
+    def passwd(self, name: str | int, service: str | None = None) -> PasswdEntry | None:
```
Comment (pbrezina) [inline sssd_test_framework/utils/tools.py:544]: Please, add the service parameter as keyword only. Same on other places.

**Comment by pbrezina** [comment]
Also please rename the commit subject to "tools: add service parameter to GetentUtils" to follow the "area: what" naming convention.

**Comment by pbrezina** [comment]
And rebase it on top of current master, initgroups call was added to getent.

File: sssd_test_framework/utils/tools.py [hunk 2]
```diff
@@ -599,54 +599,64 @@ def __init__(self, host: MultihostHost) -> None:
...
         :rtype: NetgroupEntry | None
         """
-        return self.__exec(NetgroupEntry, "netgroup", name)
+        return self.__exec(NetgroupEntry, "netgroup", name, service)
+
+    def __exec(self, cls, cmd: str, name: str | int, service=None) -> Any:
+        if service is not None:
+            service = "-s " + service
+
+        command = self.host.ssh.exec(["getent", cmd, service, name], raise_on_error=False)
```
Comment (pbrezina) [inline sssd_test_framework/utils/tools.py:658]: If service is given then the command is `getent passwd None name`. Please test your code.

**Review by pbrezina — CHANGES_REQUESTED** [review]

File: sssd_test_framework/utils/tools.py [hunk 3]
```diff
@@ -599,54 +599,64 @@ def __init__(self, host: MultihostHost) -> None:
         """
         super().__init__(host)
 
-    def passwd(self, name: str | int) -> PasswdEntry | None:
+    def passwd(self, name: str | int, service=None) -> PasswdEntry | None:
         """
         Call ``getent passwd $name``
 
-        :param name: User name or id.
+        :param name: Username or id.
```
Comment (pbrezina) [inline sssd_test_framework/utils/tools.py:606]: I don't disagree with this change, but it is not related to this commit, please move it to standalone commit if you want to keep it. Same on other places.

File: sssd_test_framework/utils/tools.py [hunk 4]
```diff
@@ -599,27 +599,31 @@ def __init__(self, host: MultihostHost) -> None:
         """
         super().__init__(host)
 
-    def passwd(self, name: str | int) -> PasswdEntry | None:
+    def passwd(self, name: str | int, service="") -> PasswdEntry | None:
```
Comment (pbrezina) [inline sssd_test_framework/utils/tools.py:602]: Keep the default value as None, it will be neater, that is `service: str | None = None`

File: sssd_test_framework/utils/tools.py [hunk 5]
```diff
@@ -599,27 +599,31 @@ def __init__(self, host: MultihostHost) -> None:
...
         """
         Call ``getent passwd $name``
 
         :param name: User name or id.
         :type name: str | int
+        :param service: Service used.
+        :type service: str | Defaults to None, optional
```
Comment (pbrezina) [inline sssd_test_framework/utils/tools.py:609]: move "defaults to None" to param, type is type

I'm don't know what editor do you use. vscode has autodocstring plugin that does this for you.

File: sssd_test_framework/utils/tools.py [hunk 6]
```diff
@@ -634,19 +638,27 @@ def initgroups(self, name: str) -> InitgroupsEntry:
...
         :return: Netgroup data, None if not found
         :rtype: NetgroupEntry | None
         """
-        return self.__exec(NetgroupEntry, "netgroup", name)
+        return self.__exec(NetgroupEntry, "netgroup", name, service)
+
+    def __exec(self, cls, cmd: str, name: str | int, service="") -> Any:
+        getent = ["getent", cmd, name]
```
Comment (pbrezina) [inline sssd_test_framework/utils/tools.py:655]: You can use:

```python
args = []

if service is not None:
    args = ["-s", service]

command = self.host.ssh.exec(["getent", *args, cmd, name], raise_on_error=False)
```

**Review by pbrezina — CHANGES_REQUESTED** [review]

File: sssd_test_framework/utils/tools.py (hunk 5 above)
Comment (sidecontrol) [reply sssd_test_framework/utils/tools.py:609]: I use pycharms.

File: sssd_test_framework/utils/tools.py [hunk 7]
```diff
@@ -599,27 +599,31 @@ def __init__(self, host: MultihostHost) -> None:
...
         super().__init__(host)
 
-    def passwd(self, name: str | int) -> PasswdEntry | None:
+    def passwd(self, name: str | int, service: str | None = None) -> PasswdEntry | None:
         """
         Call ``getent passwd $name``
 
         :param name: User name or id.
         :type name: str | int
+        :param service: Service used. | Defaults to None
```
Comment (pbrezina) [inline sssd_test_framework/utils/tools.py:608]: Service used, defaults to None.

File: sssd_test_framework/utils/tools.py [hunk 8]
```diff
@@ -634,19 +638,27 @@ def initgroups(self, name: str) -> InitgroupsEntry:
...
-        return self.__exec(NetgroupEntry, "netgroup", name)
+        return self.__exec(NetgroupEntry, "netgroup", name, service)
+
+    def __exec(self, cls, cmd: str, name: str | int, service: str | None = None) -> Any:
+        args = []
+
+        if service is not None:
+            args = ["-s", service]
+
+        command = self.host.ssh.exec(["getent", *args, cmd, name], raise_on_error=False)
```
Comment (pbrezina) [inline sssd_test_framework/utils/tools.py:661]: Please, remove this blank line. It is good to keep assignement and its check together.

**Review by pbrezina — CHANGES_REQUESTED** [review]

**Comment by pbrezina** [comment]
I added this diff and will merge it:

```diff
diff --git a/sssd_test_framework/utils/tools.py b/sssd_test_framework/utils/tools.py
index f9ae36e..284a1f7 100644
--- a/sssd_test_framework/utils/tools.py
+++ b/sssd_test_framework/utils/tools.py
@@ -599,7 +599,7 @@ class GetentUtils(MultihostUtility[MultihostHost]):
         """
         super().__init__(host)
 
-    def passwd(self, name: str | int, service: str | None = None) -> PasswdEntry | None:
+    def passwd(self, name: str | int, *, service: str | None = None) -> PasswdEntry | None:
         """
         Call ``getent passwd $name``
 
@@ -612,7 +612,7 @@ class GetentUtils(MultihostUtility[MultihostHost]):
         """
         return self.__exec(PasswdEntry, "passwd", name, service)
 
-    def group(self, name: str | int, service: str | None = None) -> GroupEntry | None:
+    def group(self, name: str | int, *, service: str | None = None) -> GroupEntry | None:
         """
         Call ``getent group $name``
 
@@ -625,7 +625,7 @@ class GetentUtils(MultihostUtility[MultihostHost]):
         """
         return self.__exec(GroupEntry, "group", name, service)
 
-    def initgroups(self, name: str) -> InitgroupsEntry:
+    def initgroups(self, name: str, *, service: str | None = None) -> InitgroupsEntry:
         """
         Call ``getent initgroups $name``
 
@@ -633,12 +633,14 @@ class GetentUtils(MultihostUtility[MultihostHost]):
 
         :param name: User name.
         :type name: str
+        :param service: Service used, defaults to None
+        :type service: str | None
         :return: Initgroups data
         :rtype: InitgroupsEntry
         """
-        return self.__exec(InitgroupsEntry, "initgroups", name)
+        return self.__exec(InitgroupsEntry, "initgroups", name, service)
 
-    def netgroup(self, name: str, service: str | None = None) -> NetgroupEntry | None:
+    def netgroup(self, name: str, *, service: str | None = None) -> NetgroupEntry | None:
         """
         Call ``getent netgroup $name``
 
@@ -655,8 +657,8 @@ class GetentUtils(MultihostUtility[MultihostHost]):
         args = []
         if service is not None:
             args = ["-s", service]
-        command = self.host.ssh.exec(["getent", *args, cmd, name], raise_on_error=False)
 
+        command = self.host.ssh.exec(["getent", *args, cmd, name], raise_on_error=False)
         if command.rc != 0:
             return None
 

```

**Review by pbrezina — APPROVED** [review]
